### PR TITLE
feature/pushNotifications

### PR DESCRIPTION
**Pull request by ddanilyuk**

### CODE REVIEW

File: Owl/Modules/Main/Chat/ChatStore.swift [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 import ComposableArchitecture
 
+var openedChatId: String?
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Main/Chat/ChatStore.swift]: What is it? Move it to more appropriate place

File: Owl/Modules/Root/Clients/PushNotification/PushNotificationClientLive.swift [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                }
+                let data = try JSONSerialization.data(
+                    withJSONObject: json,
+                    options: [.fragmentsAllowed]
+                )
+
+                let push = try JSONDecoder.customFirestore.decode(Push.self, from: data)
+                if push.chat.id == openedChatId {
+                    completionHandler([])
+                    // TODO: Move to Chat view
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Root/Clients/PushNotification/PushNotificationClientLive.swift]: 👍

File: Owl/Modules/Root/Clients/PushNotification/PushNotificationClientLive.swift [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    ) -> Effect<PushRoute, NSError> {
+        Effect.result {
+            return Result<PushRoute, Error> {
+                guard let json = response.notification.request.content.userInfo as? [String: Any] else {
+                    throw NSError(domain: "Invalid user info", code: 1)
+                }
+                let data = try JSONSerialization.data(withJSONObject: json, options: [.fragmentsAllowed])
+                let push = try JSONDecoder.customFirestore.decode(Push.self, from: data)
+                completionHandler()
+                // Only one push route
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Root/Clients/PushNotification/PushNotificationClientLive.swift]: ???

File: Owl/Modules/Root/Clients/UserDefaults/UserDefaultsClientLive.swift [hunk 4]
```diff
@@ -1,55 +1,48 @@
 //
-//  UserDefaultsClient.swift
+//  UserDefaultsClientLive.swift
 //  Owl
 //
-//  Created by Anastasia Holovash on 10.04.2022.
+//  Created by Denys Danyliuk on 04.05.2022.
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Root/Clients/UserDefaults/UserDefaultsClientLive.swift]: Revert this changes

File: Owl/Modules/Root/Clients/Validation/ValidationClientLive.swift [hunk 5]
```diff
@@ -1,35 +1,31 @@
 //
-//  ValidationClient.swift
+//  ValidationClientLive.swift
 //  Owl
 //
-//  Created by Denys Danyliuk on 15.04.2022.
+//  Created by Denys Danyliuk on 04.05.2022.
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Root/Clients/Validation/ValidationClientLive.swift]: Revert this changes

File: Owl/Modules/Root/Clients/User/UserClientLive.swift [hunk 6]
```diff
@@ -1,48 +1,40 @@
 //
-//  UserClient.swift
+//  UserClientLive.swift
 //  Owl
 //
-//  Created by Denys Danyliuk on 18.04.2022.
+//  Created by Denys Danyliuk on 04.05.2022.
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Root/Clients/User/UserClientLive.swift]: Revert this changes

**Review by AnastasiaHolovash — APPROVED** [review]

File: Owl/Modules/Root/Clients/PushNotification/PushNotificationClientLive.swift (hunk 2 above)
Comment (ddanilyuk) [reply Owl/Modules/Root/Clients/PushNotification/PushNotificationClientLive.swift]: I add ticket for this
